rollouts/pkg/packagediscovery/packagediscovery.go

```diff
@@ -149,29 +260,69 @@ func (d *PackageDiscovery) getPackagesForRepository(gitHubClient *github.Client,
 		directories := filterByPattern(selector.Directory, allDirectories)
 
 		for _, directory := range directories {
-			thisDiscoveredPackage := DiscoveredPackage{Org: selector.Org, Repo: repoName, Revision: selector.Revision, Directory: directory}
+			thisDiscoveredPackage := DiscoveredPackage{
+				Revision:   selector.Revision,
+				Directory:  directory,
+				GitHubRepo: repo,
+				Branch:     branch,
+			}
 			discoveredPackages = append(discoveredPackages, thisDiscoveredPackage)
 		}
 	} else {
-		thisDiscoveredPackage := DiscoveredPackage{Org: selector.Org, Repo: repoName, Revision: selector.Revision, Directory: selector.Directory}
+		thisDiscoveredPackage := DiscoveredPackage{
+			Revision:   selector.Revision,
+			Directory:  selector.Directory,
+			GitHubRepo: repo,
+			Branch:     branch,
+		}
 		discoveredPackages = append(discoveredPackages, thisDiscoveredPackage)
 	}
 
 	return discoveredPackages, nil
 }
 
-func (d *PackageDiscovery) getGitHubClient(ctx context.Context, selector gitopsv1alpha1.GitHubSelector) (*github.Client, error) {
+func (d *PackageDiscovery) useCache(config gitopsv1alpha1.PackagesConfig) bool {
+	return d.cache != nil && cmp.Equal(config, d.cache.config) && time.Now().Before(d.cache.expiration)
+}
+
+func (d *PackageDiscovery) NewGitLabClient(ctx context.Context, gitlabSource gitopsv1alpha1.GitLabSource) (*gitlab.Client, error) {
+
+	// initialize a gitlab client
+	secretName := gitlabSource.SecretRef.Name
+	if secretName == "" {
+		return nil, fmt.Errorf("GitLab secret reference is missing from the config")
```

Comment:
super nit: in this error message, we are camel-casing `GitLab`, but in later error messages, we leave it all lowercase (`gitlab`). I think we should pick one for consistency

Reply:
Good catch, using `lower-case` everywhere for now.